cl_transforms_stamped/src/transform-interface.lisp

```diff
@@ -46,4 +46,234 @@ to avoid a run-time error."))
                                      &key target-frame point timeout use-current-ros-time)
   (:documentation "Transforms `point' from its own frame into `target-frame'."))
 
+;; note: "with multiple latched publishers within the same node only
+;; one will latch correctly.".
+;; roslisp runs into a single ros node!
+;; this is the main reason why the roslisp node must have a unique publisher.
 
+(defparameter *transform-broadcaster* nil
+  "package variable holding the unique transform stamped broadcaster
+  of a roslisp node.")
+
+(defclass transform-broadcaster ()
+
+  ;; note: so far there is no need to directly access these internal variables.
+
+  (;; the unique roslisp transform plublisher.
+   ;; note: the roslisp node can have
+   ;; only one latched publisher working correctly
+   (transform-publisher)
+
+   ;; protects the usage of the unique roslisp transform publisher.
+   (transform-publisher-mutex))
+
+  (:documentation "simple transform stamped broadcaster.
+  it publishes the given transforms stamped on /tf topic"))
+
+(defmethod initialize-instance :after ((transform-broadcaster transform-broadcaster) &key)
+  "initializes the fields of the newly created `transform-broadcaster' object."
+
+  (setf (slot-value transform-broadcaster 'transform-publisher)
+        (roslisp:advertise "/tf" "tf2_msgs/TFMessage"))
+
+  (setf (slot-value transform-broadcaster 'transform-publisher-mutex)
+        (make-instance 'sb-thread:mutex)))
+
+(defun get-transform-broadcaster ()
+  "returns the package variable `*transform-broadcaster*'.
+  it represents the singleton of `*transform-broadcaster*'. "
+
+  (or *transform-broadcaster*
+      (setf *transform-broadcaster*
+            (make-instance 'transform-broadcaster))))
+
+(defun destroy-transform-broadcaster ()
+  "destroys the package variable `*transform-broadcaster*' and
+  unadvertises the topic /tf."
+
+  ;; todo(lisca): consider other cases too.
+  (setf *transform-broadcaster* nil)
+
+  (roslisp:unadvertise "/tf"))
+
+(defun broadcast-transforms (transform-broadcaster &rest transforms-stamped)
```

Comment:
@gheorghelisca, in my opinion, the `transform-interface` is more for the transform listeners. I think, it would be clearer to have the broadcaster functionality in a separate file. I don't mind, though, if you think otherwise.